packages/gatsby/src/schema/__tests__/__snapshots__/print.js.snap

```diff
@@ -278,6 +278,14 @@ type OneMoreTest implements Node @dontInfer {
 
 union ThisOrThat = AnotherTest | OneMoreTest
 
+type Inline {
+  foo: Nested
+}
+
+input Baz {
+  qux: Boolean
+}
+
```

Comment:
This seems to be change (regression) to schema printing? Those are defined as part of https://github.com/gatsbyjs/gatsby/blob/d86aadd692831c5ca28889eff59740c1fc3ab141/packages/gatsby/src/schema/__tests__/print.js#L99-L122

And this particular test should exclude `InlineTest` type that include those inline types:
https://github.com/gatsbyjs/gatsby/blob/d86aadd692831c5ca28889eff59740c1fc3ab141/packages/gatsby/src/schema/__tests__/print.js#L206-L216

And those types are not referenced anywhere else (afaik) - so they are "dead" types (?)

I personally don't know if they should show up or not (if the change of behaviour is desired) - I'll leave this up to you ;)

Reply:
The problem is that `graphql-compose` had changed the order of type traversal. It visits nested types too early now. So those types get added to the schema before the original type is filtered out.

To make it work the same as it used to work is expensive - we'll have to traverse all types and fields and search for usage of every type. For me, this doesn't worth the effort at the moment as it shouldn't affect the actual outcome for the end-user.

Reply:
I think given how expensive that would be it would be fine to have this "quirk" (as long as we are aware of it and it's not something we missed :) )